test/unit/org/apache/cassandra/index/sai/metrics/QueryMetricsTest.java

```diff
@@ -255,13 +261,20 @@ public void testKDTreePostingsQueryMetricsWithSingleIndex()
         assertEquals(rowsWritten, actualRows);
 
         assertTrue(((Number) getMetricValue(objectName("NumPostings", keyspace, table, v1Index, "KDTreePostings"))).longValue() > 0);
+        waitForHistogramCountEquals(objectNameNoIndex("KDTreePostingsNumPostings", keyspace, table, PER_QUERY_METRIC_TYPE), 1);
+        waitForHistogramMeanBetween(objectNameNoIndex("KDTreePostingsNumPostings", keyspace, table, PER_QUERY_METRIC_TYPE), 1.0, 1.0);
 
-        waitForVerifyHistogram(objectNameNoIndex("KDTreePostingsNumPostings", keyspace, table, PER_QUERY_METRIC_TYPE), 1);
+        // the query performed no skips, but the metric should be updated because the index was used, so we should get
+        // a single entry in the histogram with 0 skips
+        waitForHistogramCountEquals(objectNameNoIndex("KDTreePostingsSkips", keyspace, table, PER_QUERY_METRIC_TYPE), 1);
+        waitForHistogramMeanBetween(objectNameNoIndex("KDTreePostingsSkips", keyspace, table, PER_QUERY_METRIC_TYPE), 0.0, 0.0);
 
-        // V2 index is very selective, so it should lead the union merge process, causing V1 index to be not used at all.
-        execute("SELECT id1 FROM " + keyspace + "." + table + " WHERE v1 >= 0 AND v1 <= 1000 AND v2 = '5' ALLOW FILTERING");
+        // V2 index is very selective, so it should lead the union merge process, causing V1 index to skip/advance
+        execute("SELECT id1 FROM " + keyspace + "." + table + " WHERE v1 >= 0 AND v1 <= 1000 AND v2 IN ('5', '10', '20', '22') ALLOW FILTERING");
```

Comment:
why the switch to `IN`?

Reply:
Because I wanted more skips. This way I can see it really tracks the skips and not just bumps up the metric because the index is used or something equally stupid.